include/open62541/architecture_definitions.h

```diff
@@ -499,7 +505,7 @@ static UA_INLINE size_t
 UA_atomic_subSize(volatile size_t *addr, size_t decrease) {
 #if UA_MULTITHREADING >= 100
 #ifdef _MSC_VER /* Visual Studio */
-    return _InterlockedExchangeAdd(addr, - (long long) decrease) - decrease;
+    return InterlockedExchangeAdd((LONG*) addr, (LONG) (- (long long) decrease)) - decrease;
```

Comment:
Why a double cast? If you need to cast this to `LONG`, then probably
```suggestion
    return InterlockedExchangeAdd((LONG*) addr, - (LONG) decrease) - decrease;
```
should work just fine?

Then again, the original line was
```c++
return _InterlockedExchangeSub(addr, decrease) - decrease;
```
before we patched it, so it would probably make more sense to use something like
```c++
return InterlockedExchangeSubtract(addr, decrease) - decrease;
```
(if there is a chance to make it work, I'm struggling to figure out what headers are needed when).